component/prometheus/exporter/gcp/gcp_test.go

```diff
@@ -0,0 +1,123 @@
+package gcp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/pkg/river"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConvertConfig(t *testing.T) {
+	type testcase struct {
+		riverCfg               string
+		expectedArgs           Arguments
+		expectedUnmarshalError string
+	}
+	for name, tc := range map[string]testcase{
+		"err no project ids": {
+			riverCfg:               ``,
+			expectedArgs:           DefaultArguments,
+			expectedUnmarshalError: "missing required attribute \"project_ids\"",
+		},
+		"err no metrics prefixes": {
+			riverCfg: `
+				project_ids = [
+					"foo",
+					"bar",
+				]
+			`,
+			expectedArgs: func() Arguments {
+				args := DefaultArguments
+				args.ProjectIDs = []string{
+					"foo",
+					"bar",
+				}
+				return args
+			}(),
+			expectedUnmarshalError: "missing required attribute \"metrics_prefixes\"",
+		},
+		"healthy all defaults": {
+			riverCfg: `
+				project_ids = [
+					"foo",
+					"bar",
+				]
+				metrics_prefixes = [
+					"pubsub.googleapis.com/snapshot",
+					"pubsub.googleapis.com/subscription/num_undelivered_messages",
+					"pubsub.googleapis.com/subscription/oldest_unacked_message_age",
+				]
+			`,
+			expectedArgs: func() Arguments {
+				args := DefaultArguments
+				args.ProjectIDs = []string{
+					"foo",
+					"bar",
+				}
+				args.MetricPrefixes = []string{
+					"pubsub.googleapis.com/snapshot",
+					"pubsub.googleapis.com/subscription/num_undelivered_messages",
+					"pubsub.googleapis.com/subscription/oldest_unacked_message_age",
+				}
+				return args
+			}(),
+			expectedUnmarshalError: "",
+		},
+		"healthy default override": {
+			riverCfg: `
+				project_ids = [
+					"foo",
+					"bar",
+				]
+				metrics_prefixes = [
+					"pubsub.googleapis.com/snapshot",
+					"pubsub.googleapis.com/subscription/num_undelivered_messages",
+					"pubsub.googleapis.com/subscription/oldest_unacked_message_age",
+				]
+				extra_filters = [
+					"pubsub.googleapis.com/subscription:resource.labels.subscription_id=monitoring.regex.full_match(\"my-subs-prefix.*\")",
```

Comment:
as mentioned in the draft PR description, the escaped double quotes aren't great and I will look into improving this experience before finalizing for review.

Reply:
After some investigation and playing around with things, will proceed with just requiring escaped double quotes for now and will ensure the docs call that out as well.

[It looks like Google's own APIs only ever expose/accept string representations of these filters as well](https://pkg.go.dev/cloud.google.com/go/monitoring@v1.15.1/apiv3/v2/monitoringpb#ListMetricDescriptorsRequest). 

I don't see a reason/huge benefit currently to maintain our own custom representation of this filter syntax other than preventing the need to escape double quotes in filter strings in River configs for GCP. Maintaining our own representation of this syntax for unmarshaling from River would be cumbersome. Further, some semi-baked River syntax which requires the user to specify where they want to place a string, and then defining the string elsewhere (i.e. printf style format strings) are not necessarily simpler to work with than escaping.

Reply:
+1 on everything you said. Let's keep the escaped-string for the time being, and if it becomes annoying we can re-view it later.